yarn-project/kv-store/src/indexeddb/array.ts

```diff
@@ -0,0 +1,118 @@
+import { type IDBPDatabase, type IDBPObjectStore } from 'idb';
+
+import { type AztecAsyncArray } from '../interfaces/array.js';
+import { type AztecIDBSchema } from './store.js';
+
+/**
+ * A persistent array backed by IndexedDB.
+ */
+export class IndexedDBAztecArray<T> implements AztecAsyncArray<T> {
+  #_db?: IDBPObjectStore<AztecIDBSchema, ['data'], 'data', 'readwrite'>;
+  #rootDB: IDBPDatabase<AztecIDBSchema>;
+  #container: string;
+  #name: string;
+
+  constructor(rootDB: IDBPDatabase<AztecIDBSchema>, name: string) {
+    this.#rootDB = rootDB;
+    this.#name = name;
+    this.#container = `array:${this.#name}`;
+  }
+
+  set db(db: IDBPObjectStore<AztecIDBSchema, ['data'], 'data', 'readwrite'> | undefined) {
+    this.#_db = db;
+  }
+
+  get db(): IDBPObjectStore<AztecIDBSchema, ['data'], 'data', 'readwrite'> {
+    return this.#_db ? this.#_db : this.#rootDB.transaction('data', 'readwrite').store;
```

Comment:
Because of the way IndexedDB works, you need a tx for everything. This convenience method either uses the one set in the class (usually by `AztecAsyncKVStore.transactionAsync`) or creates one in place!

Reply:
sounds like that'd be a helpful comment then :upside_down_face: